src/pcre2_jit_compile.c

```diff
@@ -13202,22 +13210,28 @@ SLJIT_ASSERT(common->early_fail_start_ptr <= common->early_fail_end_ptr);
 if (private_data_size > 65536)
```

Comment:
I noticed on one of your branches, you have modified the `if (private_data_size > 65536)` check.

On master, it returns `PCRE2_ERROR_NOMEMORY` but at one time you changed it to `PCRE2_ERROR_JIT_UNSUPPORTED`.

It seems odd to report "not enough memory" when you reach that private_data_size limit, since it's not an issue with the machine's RAM, but it appears to be an implementation limitation.

Reply:
I don't remember. I have change it.

Reply:
It changes the expected file:
```
-JIT compilation was not successful (no more memory)
+JIT compilation was not successful (feature is not supported by the JIT compiler)
```
Both seem not entirely correct.

Reply:
That happens regardless. Introducing a new error code would be the best.

Reply:
If it falls back regardless, then I guess it doesn't matter. Just change it to "PCRE2_ERROR_JIT_UNSUPPORTED" if it's simply a size limitation in the JIT engine.

Reply:
Ok. Doing these things could be low hanging fruits for somebody.